demos/nrf52-beacon/src/main.rs

```diff
@@ -81,19 +61,25 @@ const APP: () = {
         )
         .unwrap();
 
-        init::LateResources {
-            radio,
-            beacon,
-            beacon_timer: ctx.device.TIMER1,
-        }
+        ctx.spawn.update().ok();
+
+        init::LateResources { radio, beacon }
     }
 
     /// Fire the beacon.
-    #[task(binds = TIMER1, resources = [beacon_timer, beacon, radio])]
-    fn timer1(ctx: timer1::Context) {
-        // Acknowledge event so that the interrupt doesn't keep firing
-        ctx.resources.beacon_timer.events_compare[0].reset();
-
+    #[task(schedule = [update], resources = [beacon, radio])]
+    fn update(ctx: update::Context) {
         ctx.resources.beacon.broadcast(ctx.resources.radio);
+
+        ctx.schedule
+            .update(ctx.scheduled + 10_666_666.cycles())
+            .ok(); // about 3 times per second as the nrf52 runs with 32 MHz
```

Comment:
Hmm, the core speed is 64 MHz though?

Reply:
The clock is at 32 Mhz. I updated the comment.

Reply:
CYCCNT counts at the core frequency from what I understand, which is 64 MHz, right?

Reply:
Yes yes, you are right :facepalm: I had a blinky test with a delay of 32_000_000 and thought it must be 32 MHz as it was blinking once a second. Well that was very wrong :)

Reply:
Fixed it, sorry